crates/sui-json-rpc/src/threshold_bls_api.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::api::ThresholdBlsApiServer;
+use crate::SuiRpcModule;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fastcrypto_tbls::{mocked_dkg, tbls::ThresholdBls, types::ThresholdBls12381MinSig};
+use jsonrpsee::core::RpcResult;
+use jsonrpsee::RpcModule;
+use move_core_types::value::MoveStructLayout;
+use std::sync::Arc;
+use sui_core::authority::AuthorityState;
+use sui_json_rpc_types::SuiTBlsSignObjectCommitmentType::{ConsensusCommitted, FastPathCommitted};
+use sui_json_rpc_types::{
+    SuiCertifiedTransactionEffects, SuiTBlsSignObjectCommitmentType,
+    SuiTBlsSignRandomnessObjectResponse,
+};
+use sui_open_rpc::Module;
+use sui_types::base_types::ObjectID;
+use sui_types::crypto::construct_tbls_randomness_object_message;
+use sui_types::object::{Object, ObjectRead};
+
+pub struct ThresholdBlsApiImpl {
+    state: Arc<AuthorityState>,
+}
+
+impl ThresholdBlsApiImpl {
+    pub fn new(state: Arc<AuthorityState>) -> Self {
+        Self { state }
+    }
+
+    /// Check that the given layout represents a Randomness object.
+    fn is_randomness_object(layout: &MoveStructLayout) -> bool {
+        let MoveStructLayout::WithTypes{type_, fields: _} = layout else { return false; };
+        let prefix = "0000000000000000000000000000000000000002::randomness::Randomness";
+        type_.to_canonical_string().starts_with(prefix)
+    }
+
+    /// Get the object and check if it is a Randomness object.
+    async fn get_randomness_object(&self, object_id: ObjectID) -> RpcResult<Object> {
+        let obj_read = self
+            .state
+            .get_object_read(&object_id)
+            .await
+            .map_err(|e| anyhow!(e))?;
+        let ObjectRead::Exists(_obj_ref, obj, layout) = obj_read else {
+            Err(anyhow!("Object not found"))? };
```

Comment:
if you return Result<Object,Error> instead of RpcResult, you can use a proper error (SuiError or sui-json-rpc::Error) instead of `Err(anyhow!())?`, e.g.
https://github.com/MystenLabs/sui/blob/2a3d2853d64f539494869aac5baf0670962bf1fd/crates/sui-json-rpc/src/coin_api.rs#L40-L42

Reply:
I changed the function response and used enums for "canonical errors" and anyhow for user specific ones.